.github/workflows/build-test-deploy.yml

```diff
@@ -1014,63 +1014,148 @@ jobs:
+
+
+  frontend-firebase-preview-deploy:
+    name: "Firebase Preview channel PR deployment"
+    needs: common-deps
+    runs-on:
+      - self-hosted
+      - linux
+      - x64
+      - sre
+    concurrency:
+      group: $${github.workflow }}-frontend-nix-deploy-firebase-${{ github.ref }}
+      cancel-in-progress: true
+    container:
+      image: niteo/nixpkgs-nixos-22.05:316b762afdb9e142a803f29c49a88b4a47db80ee
     steps:
+      - name: Get branch name
+        id: branch-name
+        uses: tj-actions/branch-names@v6
       - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - run: |
+          echo "experimental-features = nix-command flakes" > /etc/nix/nix.conf
+          echo "sandbox = relaxed" >> /etc/nix/nix.conf
+          echo "narinfo-cache-negative-ttl = 0" >> /etc/nix/nix.conf
+      - name: Check if hash changes
+        if: steps.branch-name.outputs.current_branch != 'main' # PR
+        id: feStaticHashChanges
+        run: |
+          if [ `nix build --dry-run "github:ComposableFi/composable#frontend-static" --json` == `nix build --dry-run ".#frontend-static" --json` ]; then
+            echo "::set-output name=hashChange::false"; 
+          else echo "::set-output name=hashChange::true"; 
+          fi
+
+      - uses: cachix/cachix-action@f5f67badd061acb62b5c6e25e763572ca8317004
         with:
-          node-version: 16
-          always-auth: true
-          registry-url: 'https://registry.npmjs.org'
-      - run: yarn install && yarn export --filter=picasso
-        working-directory: frontend
-        env:
-          SUBSTRATE_PROVIDER_URL_KUSAMA_2019: ${{ secrets.SUBSTRATE_PROVIDER_URL_KUSAMA_2019 }}
-          SUBSTRATE_PROVIDER_URL_KUSAMA: ${{ secrets.SUBSTRATE_PROVIDER_URL_KUSAMA }}
-          SUBSTRATE_PROVIDER_URL_KARURA: ${{ secrets.SUBSTRATE_PROVIDER_URL_KARURA }}
-          SUBSQUID_URL: ${{ secrets.SUBSQUID_URL }}
-          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-      - uses: FirebaseExtended/action-hosting-deploy@0f248036885ae672f78587669fa2923f0ef6cac2
+          skipPush: true
+          installCommand: |
+            nix-channel --add ${{env.NIX_NIXPKGS_CHANNEL }} nixpkgs
+            nix-channel --update
+            nix-env -iA nixpkgs.cachix
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+          name: composable-community
+      - run: |
+          cachix watch-exec --jobs 16 --compression-level 0 composable-community nix -- build .#frontend-static-firebase --no-update-lock-file --show-trace -L
+          mkdir -p firebase/picasso/out/
+          mkdir -p firebase/pablo/out/
+          cp -R result/picasso/* firebase/picasso/out/
+          cp -R result/pablo/* firebase/pablo/out/
+          cp frontend/apps/pablo/.firebaserc firebase/pablo/
+          cp frontend/apps/pablo/firebase.json firebase/pablo/
+          cp frontend/apps/picasso/.firebaserc firebase/picasso/
+          cp frontend/apps/picasso/firebase.json firebase/picasso/
```

Comment:
nit: would be nicer if `.#frontend-static-firebase` would leave the result directory in a deployable state (so handling the copying of the `firebase.json` and `.firebaserc` there. Then this flow can simply set the `entrypoint: result/pablo` `entrypoint: result/picasso`